test/liblangutil/Scanner.cpp

```diff
@@ -123,7 +123,14 @@ BOOST_AUTO_TEST_CASE(string_nonprintable)
 		if (v == '\n' || v == '\v' || v == '\f' || v == '\r')
 			BOOST_CHECK_EQUAL(scanner.currentError(), ScannerError::IllegalStringEndQuote);
 		else
-			BOOST_CHECK_EQUAL(scanner.currentError(), ScannerError::IllegalCharacterInString);
+		{
+			BOOST_CHECK_EQUAL(
+				scanner.currentError(),
+				(scanner.scannerKind() == ScannerKind::Solidity) ?
```

Comment:
Does this actually ever run with ``scanner.scannerKind() == ScannerKind::Yul``?

Reply:
It does not. I changed it in the tests and they're still running as expected.